src/DGtal/shapes/MeshVoxelizer.h

```diff
@@ -109,15 +109,111 @@ namespace DGtal
     {
       PointR3 myFirst, mySecond;
       Edge() {}
-      Edge(PointR3 mf, PointR3 ms) : myFirst(mf), mySecond(ms) {}
+      Edge(const PointR3& mf, const PointR3& ms) : myFirst(mf), mySecond(ms) {}
+      Edge(const Edge&) = default;
+    };
+
+    /**
+     * Internal Intersection Target structure
+     *
+     * @tparam Dimension dimension of the intersection target
+     * @tparam Separation strategy of the voxelization (6 or 26)
+     * @tparam Dummy dummy parameter just to be able to specialize inside template class (FIXME with another location)
+     */
+    template<size_t Sep, size_t Dimension = 1, typename Dummy = void>
+    struct IntersectionTarget { };
+
+    /**
+     * Specialization for 6-separated with 1D intersection target
+     */
+    template<typename Dummy>
+    struct IntersectionTarget<6, 1, Dummy>
```

Comment:
Agree with an external "intersectiontarget.h" file (same place). When defining the target, please consider integer only computations as far as possible.  (like 0.5-> 1 and "/2" if needed).